### PR TITLE
[KYUUBI #3154][Subtask] Connectors for Spark SQL Query Engine -> TiDB/TiKV

### LINKED ISSUE #3154: [Subtask] Connectors for Spark SQL Query Engine -> TiSpark
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

Add a doc of Spark SQL Query Engine for TiSpark.

### Parent issue

#3059

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhouyifan279**
### _Why are the changes needed?_
Close #3154

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zhouyifan279** [comment]
cc @yaooqinn @pan3793

**Review by cxzl25 — APPROVED** [review]

File: docs/connector/spark/tispark.rst [hunk 1]
```diff
@@ -16,21 +16,86 @@
...
+   SELECT * FROM foo;
+
+Taking ``DELETE FROM`` as a example, Spark 3 added support for DELETE FROM queries to remove data from tables.
+
+.. code-block:: sql
+
+   DELETE FROM foo WHERE id >= 1 and id < 2;
+
+.. note::
+   As for now (TiSpark 3.0.1), TiSpark does not support ``CREATE TABLE``, ``INSERT INTO/OVERWRITE``` operations
```
Comment (zhaomin1423) [inline docs/connector/spark/tispark.rst:96]: After `***/OVERWRITE`, there is a redundant `.
image

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
we can rename the title from `TiSpark`_ to TiDB/TiKv？

**Comment by yaooqinn** [comment]
I mean the doc title

File: docs/connector/spark/tispark.rst (hunk 1 above)
Comment (zhouyifan279) [reply docs/connector/spark/tispark.rst]: Thanks, fixed.

**Comment by zhouyifan279** [comment]
> we can rename the title from `TiSpark`_ to TiDB/TiKv？
1. Changed doc title and some sub-titles.

**Review by yaooqinn — APPROVED** [review]

File: docs/connector/spark/tidb.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+   DELETE FROM foo WHERE id >= 1 and id < 2;
+
+.. note::
+   As for now (TiSpark 3.0.1), TiSpark does not support ``CREATE TABLE``, ``INSERT INTO/OVERWRITE`` operations
+   through Apache Spark Datasource V2 and Catalog APIs.
+
+.. _Official Documentation: https://docs.pingcap.com/tidb/stable/overview
+.. _Maven Central: https://repo1.maven.org/maven2/com/pingcap/tispark/
+.. _Environment setup: https://docs.pingcap.com/tidb/stable/tispark-overview#environment-setup
```
Comment (yaooqinn) [inline docs/connector/spark/tidb.rst:103]: TiSpark Environment setup

Comment (zhouyifan279) [reply docs/connector/spark/tidb.rst]: Done

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**